Failure # 1 (occurred at 2024-11-20_18-06-32)
[36mray::ImplicitFunc.train()[39m (pid=80656, ip=172.24.11.109, actor_id=8ce32ddd5be48535bc502a7b01000000, repr=train)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/kgreedy/lib/python3.12/site-packages/ray/tune/trainable/trainable.py", line 331, in train
    raise skipped from exception_cause(skipped)
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/kgreedy/lib/python3.12/site-packages/ray/air/_internal/util.py", line 104, in run
    self._ret = self._target(*self._args, **self._kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/kgreedy/lib/python3.12/site-packages/ray/tune/trainable/function_trainable.py", line 45, in <lambda>
    training_func=lambda: self._trainable_func(self.config),
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/kgreedy/lib/python3.12/site-packages/ray/tune/trainable/function_trainable.py", line 250, in _trainable_func
    output = fn()
             ^^^^
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/main.py", line 68, in train
    agent = TrainModel(
            ^^^^^^^^^^^
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/utils/agent.py", line 70, in __init__
    self._get_centroids(self._training_data, self._training_labels)
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/utils/agent.py", line 122, in _get_centroids
    sub_centroids = [np.mean(cluster, axis=0) for cluster in np.array_split(cdata, self._clusters)]
                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/kgreedy/lib/python3.12/site-packages/pennylane/numpy/wrapper.py", line 117, in _wrapped
    res = obj(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/kgreedy/lib/python3.12/site-packages/autograd/tracer.py", line 48, in f_wrapped
    return f_raw(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/digvijay/Documents/developer/greedy_kernel_alignment/kgreedy/lib/python3.12/site-packages/numpy/lib/shape_base.py", line 782, in array_split
    sub_arys.append(_nx.swapaxes(sary[st:end], axis, 0))
                                 ~~~~^^^^^^^^
TypeError: slice indices must be integers or None or have an __index__ method
